### PR TITLE
Text replace scripts

### PR DESCRIPTION
**Pull request by EYH0602**
finished `scanf` and `cin` with return.
update text replacement script #3

### CODE REVIEW

File: encode2stderr.hpp [hunk 1]
```diff
@@ -11,144 +11,136 @@
...
-        std::cerr << #x << ",," << x << "\n"; \
-    }
+#define CIN(x)                                                                 \
+  [&x]() {                                                                     \
+    std::cin >> x;                                                             \
+    if (std::cin.good()) {                                                     \
+      std::cout << #x << ",," << x << "\n";                                    \
+    }                                                                          \
+    return std::cin.good();                                                    \
+  }()\
```
Comment (EYH0602) [inline encode2stderr.hpp:146]: need `{}` here

File: encode2stderr.hpp [hunk 2]
```diff
@@ -11,144 +11,136 @@
...
+  } else if (type == "hd") {
+    fprintf(stderr, pfmt.c_str(), name.c_str(), type.c_str(),
+            *(short *)val_ptr);
+  } else if (type == "f") {
+    fprintf(stderr, pfmt.c_str(), name.c_str(), type.c_str(),
+            *(float *)val_ptr);
+  } else if (type == "lf" || type == "Lf") {
+    fprintf(stderr, pfmt.c_str(), name.c_str(), type.c_str(),
+            *(double *)val_ptr);
+  } else {
```
Comment (DataCorrupted) [inline encode2stderr.hpp:107]: I recommend `else if (... == 'd') {} else { unreachable; }` to make code more clear

File: scripts/build.sh [hunk 3]
```diff
@@ -22,14 +22,18 @@ for I in $PROBLEMS; do
...
             touch $SRCDIR/$P.cpp
             cat $EMBDING_HOME/header.hpp >> $SRCDIR/$P.cpp
-            cat $TXTDIR/$P.txt >> $SRCDIR/$P.cpp
+            cat $EMBDING_HOME/encode2stderr.hpp >> $SRCDIR/$P.cpp
+            # TODO combine reokace_input into fix.py in hyf-dev
+            python3.8 $EMBDING_HOME/scripts/replace_input.py $TXTDIR/$P.txt >> $SRCDIR/$P.cpp
             sed -i 's/void main/int main/g' $SRCDIR/$P.cpp
         fi
         if [ ! -f $OUTDIR/$P ]; then
+            # TODO: check if file contains `gets()` then use --std=c++11
```
Comment (DataCorrupted) [inline scripts/build.sh:31]: Didn't we default to C++11?

File: scripts/replace_input.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    return "CIN({})".format(var.strip())
+
+
+def replace_cin(line: str) -> str:
+    tokens = line.split("cin")
+    # print leading terms
+    # for example, indentation, while(
+    # print(tokens[0], end="")
+    out_str = tokens[0]
+    is_while_loop = "while" in tokens[0]
```
Comment (EYH0602) [inline scripts/replace_input.py:20]: there are for loop also use `cin`

for example 1/1176
```cpp
for (cin >> n; n > 0; n--) 
// is replaced by
for (CIN(n; n--);
```

File: encode2stderr.hpp [hunk 5]
```diff
@@ -11,144 +11,136 @@
...
-        std::vector<std::string> type_tokens = split_fmt_types(fmt);              \
-        void* values[] = {args};                                                  \
-        assert(name_tokens.size() == narg);                                       \
-        assert(type_tokens.size() == narg);                                       \
-                                                                                  \
-        for (int i = 0; i < narg; i++) {                                          \
-            print_stderr(name_tokens[i], type_tokens[i], values[i]);              \
-        }                                                                         \
-    }
+#define SCANF_ALT(fmt, args...) scanf_alt(fmt, #args, args)
```
Comment (EYH0602) [inline encode2stderr.hpp:136]: error if `scanf("\n");` with no other args

File: scripts/check_data_integrity.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        err_msg = "Failed for {}\n".format(dir_name) + err_msg + "\n"
+
+    return err_msg
+
+
+if __name__ == "__main__":
+    output_dir = (
+        os.environ["OUTPUT"]
+        if os.getenv("OUTPUT")
+        else os.environ["EMBDING_HOME"] + "/output-1"
```
Comment (DataCorrupted) [inline scripts/check_data_integrity.py:61]: import common, `EMBDING_HOME`, and `CORES` can be directly used.

File: encode2stderr.hpp [hunk 7]
```diff
@@ -11,144 +11,143 @@
...
+      std::cout << #x << ",," << x << "\n";                                    \
+    }                                                                          \
+    return std::cin.good();                                                    \
+  }()
+
+// macro for cin
+#define CIN(x)                                                                 \
+  {                                                                            \
+    std::cin >> x;                                                             \
+    std::cout << #x << ",," << x << "\n";                                      \
```
Comment (DataCorrupted) [inline encode2stderr.hpp:152]: cerr

File: encode2stderr.hpp [hunk 8]
```diff
@@ -11,144 +11,143 @@
...
-#define CIN(x)                                \
-    {                                         \
-        std::cin >> x;                        \
-        std::cerr << #x << ",," << x << "\n"; \
-    }
+#define CIN_LOOP(x)                                                            \
+  [&x]() {                                                                     \
+    std::cin >> x;                                                             \
+    if (std::cin.good()) {                                                     \
+      std::cout << #x << ",," << x << "\n";                                    \
```
Comment (DataCorrupted) [inline encode2stderr.hpp:143]: cerr

File: encode2stderr.hpp [hunk 9]
```diff
@@ -11,144 +11,143 @@
...
+    fprintf(stderr, pfmt.c_str(), name.c_str(), type.c_str(),
+            *(short *)val_ptr);
+  } else if (type == "f") {
+    fprintf(stderr, pfmt.c_str(), name.c_str(), type.c_str(),
+            *(float *)val_ptr);
+  } else if (type == "lf" || type == "Lf") {
+    fprintf(stderr, pfmt.c_str(), name.c_str(), type.c_str(),
+            *(double *)val_ptr);
+  } else {
+    // set default type to int
```
Comment (DataCorrupted) [inline encode2stderr.hpp:108]: remove comments

**Review by DataCorrupted — CHANGES_REQUESTED** [review]

**Review by DataCorrupted — APPROVED** [review]

**Pull request merged**